game-core/src/main/java/org/triplea/debug/error/reporting/StackTraceErrorReportFormatter.java

```diff
@@ -91,18 +92,15 @@ private String buildBody(@Nullable final String userDescription, final LogRecord
   }
 
   private static String throwableToString(final Throwable e) {
+    final var outputStream = new ByteArrayOutputStream();
+    try (PrintWriter printWriter = new PrintWriter(outputStream, false, StandardCharsets.UTF_8)) {
```

Comment:
- How do you know this works @RoiEXLab ? Are you going off of Stack Overflow, are there other examples in the code, or is this a pattern you just know?
- Can you simulate an exception going through this and give an example? Perhaps better yet, evaluate this to a strategy object and then test it. (EG:  this method becomes a `Function`, then testing the function you can verify that key words from a nested exception are all presented/printed)

Reply:
1. It's just a pattern I know. Basically I just looked at the overloads of `Throwable#printStackTrace`, checked the `PrintWriter` constructor and added the `ByteArrayOutputStream` that would store it in a string for me.
2. There are already test cases in place that check for the existence of some lines:
https://github.com/triplea-game/triplea/blob/b133fd9387cc6d4295e6f2d71d193e2ff8e8b977/game-core/src/test/java/org/triplea/debug/error/reporting/StackTraceErrorReportFormatterTest.java#L137-L150
3. I just realized I should maybe extend this test case so it verifies the causes are also checked

Reply:
Thanks, good work @RoiEXLab